### PR TITLE
Desing Patterns by Deyan Sadinov

### PR DESCRIPTION
**Pull request by deyansadinov**
Observer Pattern - task5

### CODE REVIEW

File: src/task5/observerpattern/Observer.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package task5.observerpattern;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public interface Observer {
```
Comment (mgenov) [inline src/task5/observerpattern/Observer.java]: Same as: https://github.com/tkehayov/design-patterns/blob/observer/src/com/clouway/Observer.java

File: src/task5/observerpattern/StockGrabber.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package task5.observerpattern;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class StockGrabber implements Subject {
```
Comment (mgenov) [inline src/task5/observerpattern/StockGrabber.java]: Same as: 

https://github.com/tkehayov/design-patterns/blob/observer/src/com/clouway/StockGrabber.java

File: src/task5/observerpattern/Subject.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package task5.observerpattern;
+
+
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public interface Subject {
+
+  public void register(Observer o);
```
Comment (mgenov) [inline src/task5/observerpattern/Subject.java]: public modifiers are not required

File: src/newtask5/NotAvailableProducts.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package newtask5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Deyan Sadinov <sadinov88@gmail.com>
+ */
+public class NotAvailableProducts implements ProductListener {
```
Comment (mgenov) [inline src/newtask5/NotAvailableProducts.java]: Looks like a bad name for implementation. Think for a better one.

Comment (deyansadinov) [inline src/newtask5/NotAvailableProducts.java]: I renamed the implementation

**Comment by mgenov** [comment]
LGTM !

**Pull request merged**